### PR TITLE
Copy entire build output before publish

### PR DESCRIPTION
**Pull request by zijchen**
Address #25517 

Recent change in DacFx removed full paths of references from build output. Because ADS copies the dacpac to a temp location before publish (due to a locking issue #11036), this is causing the referenced dacpac to not be found. Fix is to copy the entire output folder which should contain all the necessary references.

### CODE REVIEW

**Review by Charles-Gagnon — APPROVED** [review]
Make sure to update the extension version as well

File: extensions/sql-database-projects/src/controllers/projectController.ts [hunk 1]
```diff
@@ -540,9 +540,11 @@ export class ProjectsController {
 			return undefined; // buildProject() handles displaying the error
 		}
 
-		// copy dacpac to temp location before publishing
-		const tempPath = path.join(os.tmpdir(), `${path.parse(dacpacPath).name}_${new Date().getTime()}${constants.sqlprojExtension}`);
-		await fs.copyFile(dacpacPath, tempPath);
+		// copy entire build output to temp location before publishing
+		const tempDir = path.join(os.tmpdir(), `${path.parse(dacpacPath).name}_${new Date().getTime()}`);
+		await fs.mkdir(tempDir);
```
Comment (ssreerama) [inline extensions/sql-database-projects/src/controllers/projectController.ts:545]: What are we doing the Temp directory after the publishing done? Are we cleaning it here or in DacFx? Thanks

**Review by ssreerama — APPROVED** [review]

**Review by kisantia — APPROVED** [review]

**Comment by kburtram** [comment]
@zijchen is this change still planned or should this be closed as abandoned?

**Comment by zijchen** [comment]
> @zijchen is this change still planned or should this be closed as abandoned?

We decided to revert the change in DacFx which was why I left this PR idle, but I think it's still good to have because there could be other artifacts the dacpac requires and expects in the same folder. I'll merge it.

**Pull request merged**